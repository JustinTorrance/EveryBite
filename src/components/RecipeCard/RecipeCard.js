import React from 'react'
import CardActions from '@material-ui/core/CardActions'
import Button from '@material-ui/core/Button'
import shortid from 'shortid'
import PropTypes from 'prop-types'

const RecipeCard = ({name, id, image, ingredients, time}) => {

 const mappedIngredients = ingredients.map(ingredient => {
   return <p className='ingredient-p' key={shortid.generate()}>{ingredient}</p>
 })

 const minutes = parseInt(time) / 60

  return(
      <article className='card'>
        <h3 className='recipe-name'>{name}</h3>
        <p className='time-p'>Time: {minutes} minutes</p>
        <img src={image} className='img' alt={`${name}`} />
        <div className='ingredients-div'>
          <span className='ingredient-label'>Ingredients:</span> {mappedIngredients}
        </div>
        <div className='info-btn'>
          <CardActions>
            <Button size="small" color="primary" href={`https://www.yummly.com/recipe/${id}`} target="_blank">
                More Info
            </Button>
          </CardActions>
        </div>
      </article>
  )
}

RecipeCard.propTypes = {
  name: PropTypes.string,
  time: PropTypes.number,
  id: PropTypes.string,
  image: PropTypes.string,
  ingredients: PropTypes.array
}

export default RecipeCard