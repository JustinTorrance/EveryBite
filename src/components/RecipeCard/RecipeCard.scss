.card {
  background:#7f7f7f;
  background:rgba(255,255,255,0.5);
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding-top: 10px;
  font-family: 'Roboto', sans-serif;
  padding: 10px 10px 0 10px;
}

.recipe-name {
  text-align: center;
  margin: 0 8px;
}

.img {
  width: 100%;
  border-radius: 5px;
  margin: 10px 0;
}

.info-btn {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: flex-end;
}

.ingredients-div {
  width: 100%;
  padding-left: 10px;
}

.ingredient-label {
  font-weight: 900;
  font-size: 1.1rem;
}

.ingredient-p {
  margin: 2px 0;
}

.recipe-name, .time-p {
  margin: 5px 0;
}

.recipe-name {
  font-size: 1.4rem;
}


