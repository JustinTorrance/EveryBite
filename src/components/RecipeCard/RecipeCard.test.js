import React from 'react';
import { shallow } from 'enzyme';
import RecipeCard from './RecipeCard';

describe('RecipeCard', () => {
  const wrapper = shallow(
  <RecipeCard 
    name={'curry'}
    time={60}
    id={'1'}
    image={'image'}
    ingredients={['rice', 'beans']}

  />);

  it('should match the snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});