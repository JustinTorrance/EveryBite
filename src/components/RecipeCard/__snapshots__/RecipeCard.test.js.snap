// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard should match the snapshot 1`] = `
"<article className=\\"card\\">
  <h3 className=\\"recipe-name\\">
    curry
  </h3>
  <p className=\\"time-p\\">
    Time: 
    1
     minutes
  </p>
  <img src=\\"image\\" className=\\"img\\" alt=\\"curry\\" />
  <div className=\\"ingredients-div\\">
    <span className=\\"ingredient-label\\">
      Ingredients:
    </span>
     
    <p className=\\"ingredient-p\\">
      rice
    </p>
    <p className=\\"ingredient-p\\">
      beans
    </p>
  </div>
  <div className=\\"info-btn\\">
    <WithStyles(CardActions)>
      <WithStyles(Button) size=\\"small\\" color=\\"primary\\" href=\\"https://www.yummly.com/recipe/1\\" target=\\"_blank\\">
        More Info
      </WithStyles(Button)>
    </WithStyles(CardActions)>
  </div>
</article>"
`;
